lib/src/Pos/Logic/Full.hs

```diff
@@ -0,0 +1,246 @@
+                (\(MCCommitment (pk, _, _)) -> addressHash pk)
+                (\id tm -> MCCommitment <$> tm ^. tmCommitments . to getCommitmentsMap . at id)
+                (\(MCCommitment comm) -> sscProcessCommitment comm)
+
+            postSscOpening = postSscCommon
+                OpeningMsg
+                (\(MCOpening key _) -> key)
+                (\id tm -> MCOpening id <$> tm ^. tmOpenings . at id)
+                (\(MCOpening key open) -> sscProcessOpening key open)
+
+            postSscShares = postSscCommon
+                SharesMsg
+                (\(MCShares key _) -> key)
+                (\id tm -> MCShares id <$> tm ^. tmShares . at id)
+                (\(MCShares key shares) -> sscProcessShares key shares)
+
+            postSscVssCert = postSscCommon
+                VssCertificateMsg
+                (\(MCVssCertificate vc) -> getCertId vc)
+                (\id tm -> MCVssCertificate <$> lookupVss id (tm ^. tmCertificates))
+                (\(MCVssCertificate cert) -> sscProcessCertificate cert)
+
+            postSscCommon
+                :: ( Buildable err, Buildable contents )
+                => SscTag
+                -> (contents -> StakeholderId)
+                -> (StakeholderId -> TossModifier -> Maybe contents)
+                -> (contents -> ExceptT err m ())
+                -> KeyVal (Tagged contents StakeholderId) contents m
+            postSscCommon sscTag contentsToKey toContents processData = KeyVal
+                { toKey = pure . tagWith contentsProxy . contentsToKey
+                , handleInv = sscIsDataUseful sscTag . unTagged
+                , handleReq = \(Tagged addr) -> toContents addr . view ldModifier <$> sscRunLocalQuery ask
+                , handleData = \dat -> do
+                      let addr = contentsToKey dat
+                      -- [CSL-685] TODO: Add here malicious emulation for network
+                      -- addresses when TW will support getting peer address
+                      -- properly
+                      -- Stale comment?
+                      handleDataDo dat addr =<< shouldIgnorePkAddress addr
+                }
+              where
+                contentsProxy = (const Proxy :: (contents -> k) -> Proxy contents) contentsToKey
+                ignoreFmt =
+                    "Malicious emulation: data "%build%" for id "%build%" is ignored"
+                handleDataDo dat id shouldIgnore
+                    | shouldIgnore = False <$ logDebug (sformat ignoreFmt id dat)
+                    | otherwise = sscProcessMessage processData dat
+                sscProcessMessage sscProcessMessageDo dat =
+                    runExceptT (sscProcessMessageDo dat) >>= \case
+                        Left err -> False <$ logDebug (sformat ("Data is rejected, reason: "%build) err)
+                        Right () -> return True
+
+            -- TODO implement these proper
+            -- I do not know of an existing program which will do what we want:
```

Comment:
I'm not sure what you're talking about here. There's "internal datatype consistency" verification level that you can use here, also "pure checks", but only those that don't require GState.

 In general, I assume that it's a good strategy to create issues for this kind of things instead of putting zero-functional stubs.